### ISSUE TITLE
Support SegFormer

### ISSUE THREAD

**Issue by DavidLandup0** [keras-team/keras-cv#1169]
**Short Description**

As 0.4 is soon to roll out which'll include support for semantic segmentation with DeepLabV3 (and possibly DeepLabV3+), and as we're ramping up Transformer-vision support - I propose that we also add SegFormer soon.

- Official Implementation (PyTorch): https://github.com/NVlabs/SegFormer
- Paper: https://arxiv.org/abs/2105.15203
- Citations: 670

Fewer flops and higher accuracy than DeepLabV3, and offers a family of B0 to B5 (accuracy-flops tradeoff instead of a rigid structure):

<img width="524" alt="image" src="https://user-images.githubusercontent.com/60978046/208577666-bd84f237-f857-40c5-9c92-75f3ca32164e.png">

**Other Information**

I'd love to work on this one too :)

**Comment by IMvision12** [comment]
I guess along with segformer we should also add MaskFormer

**Comment by innat** [comment]
https://github.com/deep-diver/segformer-tf-transformers

**Comment by DavidLandup0** [comment]
Thanks for the links! Sayak did the SegFormer for HuggingFace, and I really liked the implementation. It'd be nice to reference it and do a similar structure, modularizing whenever possible.

**Comment by tanzhenyu** [comment]
@sayakpaul FYI

**Comment by IMvision12** [comment]
Hey, @DavidLandup0 Are you working on adding segformer? If not Can take up this issue?

**Comment by cosmo3769** [comment]
Hey @DavidLandup0, I was working with semantic segmentation in my project and find out segformer pretty nice. I would love to work on this one too.
